c_glib/arrow-glib/buffer.cpp

```diff
@@ -357,6 +362,106 @@ garrow_mutable_buffer_slice(GArrowMutableBuffer *buffer,
   return garrow_mutable_buffer_new_raw(&arrow_buffer);
 }
 
+
+G_DEFINE_TYPE(GArrowResizableBuffer,              \
+              garrow_resizable_buffer,            \
+              GARROW_TYPE_MUTABLE_BUFFER)
+
+static void
+garrow_resizable_buffer_init(GArrowResizableBuffer *object)
+{
+}
+
+static void
+garrow_resizable_buffer_class_init(GArrowResizableBufferClass *klass)
+{
+}
+
+/**
+ * garrow_resizable_buffer_resize:
+ * @buffer: A #GArrowResizableBuffer.
+ * @new_size: The new buffer size in bytes.
+ * @error: (nullable): Return location for a #GError or %NULL.
+ *
+ * Returns: %TRUE on success, %FALSE if there was an error.
+ *
+ * Since: 0.3.0
+ */
+gboolean
+garrow_resizable_buffer_resize(GArrowResizableBuffer *buffer,
+                               gint64 new_size,
+                               GError **error)
+{
+  auto arrow_buffer = garrow_buffer_get_raw(GARROW_BUFFER(buffer));
+  auto arrow_resizable_buffer =
+    std::static_pointer_cast<arrow::ResizableBuffer>(arrow_buffer);
+  auto status = arrow_resizable_buffer->Resize(new_size);
+  if (status.ok()) {
+    return TRUE;
+  } else {
+    garrow_error_set(error, status, "[resizable-buffer][resize]");
+    return FALSE;
+  }
```

Comment:
I see this pattern a lot in the code. I wonder if you want

```c++
#define GARROW_RETURN_IF_ERROR(s, ERROR_MSG) {  \
  do {                                          \
    ::arrow::Status _s = (s);                   \
    if (!_s.ok()) {                             \
      garrow_error_set(error, _s, ERROR_MSG);   \
      return FALSE;                             \
    }                                           \
  } while(0);
```

then you can write

```c++
GARROW_RETURN_IF_ERROR(arrow_resizable_buffer->Resize(new_size), "[resizable-buffer][resize]");
return TRUE;
```